### PR TITLE
fix: [PL-32445]: Check if SAML is enabled before using the SAML logout URL

### PR DESCRIPTION
**Pull request by tiwaritejas**
## Describe your changes
SAML logoutURL should only be used when SAML is enabled for an account. Currently, it gets used solely if the SAML settings are present. This PR adds an additional check for SAML enablement in the account.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55729)

### CODE REVIEW

**Comment by tiwaritejas** [comment]
trigger uts

**Review by sahibaHarness — APPROVED** [review]

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55729)* status: 1 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java` line 2752 at r1](https://reviewable.io/reviews/harness/harness-core/55729#-NjlRUtTAVxcAk8WNZLY:-NjlRUtTAVxcAk8WNZLZ:b6g9abt) ([raw file](https://github.com/harness/harness-core/blob/da5e6ae20e7ca07d18ff03cd017725425d4acf15/400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java#L2752)):*
> ```Java
>     log.info("Sending logout response from manager for user {} in account {}", userId, accountId);
> 
>     Account account = authenticationUtils.getAccount(accountId);
> ```

account = null in case of error, so it will result into NPE in next line

**Comment by tiwaritejas** [comment]
trigger uts

**Review by tiwaritejas — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55729)* status: 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java` line 2752 at r1](https://reviewable.io/reviews/harness/harness-core/55729#-NjlRUtTAVxcAk8WNZLY:-NjlTH7W5D1ig4-CY_Z_:bqmeti3) ([raw file](https://github.com/harness/harness-core/blob/da5e6ae20e7ca07d18ff03cd017725425d4acf15/400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java#L2752)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Also, if it throws exception, then do we want to fail the complete use-case or handle it elegantly?
</blockquote></details>

Added exception handling

**Review by mohitg0795 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55729)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by VikasMaddukuriHarness — APPROVED** [review]

**Comment by tiwaritejas** [comment]
trigger messagecheck

**Comment by tiwaritejas** [comment]
trigger uts

**Comment by tiwaritejas** [comment]
trigger uts

**Review by sahibaHarness — DISMISSED** [review]

File: 400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -2747,11 +2748,19 @@ public void logout(User user) {
...
-      log.info("Logout URL from accountId is {}", samlSettings.getLogoutUrl());
+    Account account = null;
+    try {
+      account = authenticationUtils.getAccount(accountId);
+    } catch (Exception ex) {
+      log.error("Error while fetching account details for accountId {}", accountId, ex);
+    }
+    if (account != null && account.getAuthenticationMechanism() == SAML) {
+      SamlSettings samlSettings = ssoSettingService.getSamlSettingsByAccountId(accountId);
+      log.info("Samlsettings from accountId is {}", samlSettings);
```
Comment (pbarapatre10) [inline 400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java:2759]: nit: lets not log the entire saml settings, as it contain metadata xml as well, with certificate details, thanks

File: 400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -2747,11 +2748,19 @@ public void logout(User user) {
...
+      if (samlSettings != null && samlSettings.getLogoutUrl() != null) {
```
Comment (pbarapatre10) [inline 400-rest/src/main/java/software/wings/service/impl/UserServiceImpl.java:2760]: nit: more appropriate `&& isNotEmpty(samlSettings.getLogoutUrl())`, thanks

**Review by pbarapatre10 — DISMISSED** [review]

**Comment by tiwaritejas** [comment]
trigger uts

**Comment by tiwaritejas** [comment]
trigger ut6,ut7

**Comment by tiwaritejas** [comment]
trigger uts

**Review by pbarapatre10 — APPROVED** [review]

**Pull request merged**